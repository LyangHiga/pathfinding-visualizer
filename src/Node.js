import React from "react";
import "./Node.css";

export default function Node(props) {
  return (
    <div
      id={`node-${props.val}`}
      className={"node"}
      onMouseDown={() => props.onMouseDown(props.row, props.col)}
      onMouseEnter={() => props.onMouseEnter(props.row, props.col)}
      onMouseUp={() => props.onMouseUp()}
    >
      {props.isWeighted ? <span>{props.weight}</span> : ""}
    </div>
  );
}
