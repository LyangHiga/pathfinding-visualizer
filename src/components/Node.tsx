import React from "react";
import styles from "../styles/NodeStyles";

interface NodeProps {
  val: number;
  row: number;
  col: number;
  isWeighted: boolean;
  weight: number;
  onMouseDown(row: number, col: number): void;
  onMouseEnter(row: number, col: number): void;
  onMouseUp(): void;
}

export default function Node(props: NodeProps) {
  const classes = styles();
  return (
    <div
      id={`node-${props.val}`}
      className={classes.node}
      onMouseDown={() => props.onMouseDown(props.row, props.col)}
      onMouseEnter={() => props.onMouseEnter(props.row, props.col)}
      onMouseUp={() => props.onMouseUp()}
    >
      {props.isWeighted ? <span>{props.weight}</span> : ""}
    </div>
  );
}
