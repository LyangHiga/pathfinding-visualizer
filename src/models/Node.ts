import AdjList from "./AdjList";

export default class Node {
  private col: number;
  private row: number;
  // converting row x col to a single vector
  // unique id
  public val: number;
  public weight: number;
  public isStart: boolean;
  public isTarget: boolean;
  public isWall: boolean;
  public adjList: AdjList;

  constructor(
    col: number,
    row: number,
    val: number,
    start: number,
    target: number,
    adjList: AdjList,
    weight = 1
  ) {
    this.col = col;
    this.row = row;
    this.val = val;
    this.weight = weight;
    this.isStart = this.val === start;
    this.isTarget = this.val === target;
    this.isWall = false;
    this.adjList = adjList;
  }
}
