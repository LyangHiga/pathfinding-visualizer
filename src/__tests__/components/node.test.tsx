import React from "react";
import { shallow } from "enzyme";
import Node from "../../components/Node";

const handleMouseDown = (row: number, col: number) => {};
const handleMouseEnter = (row: number, col: number) => {};
const handleMouseUp = () => {};

describe("Node", () => {
  it("renders correctly", () => {
    const wrapper = shallow(
      <Node
        val={1}
        col={1}
        row={1}
        onMouseDown={handleMouseDown}
        onMouseEnter={handleMouseEnter}
        onMouseUp={handleMouseUp}
        weight={10}
        isWeighted={true}
      />
    );
    expect(wrapper).toMatchSnapshot();
    // On the first run of this test, Jest will generate a snapshot file automatically.
  });
});
