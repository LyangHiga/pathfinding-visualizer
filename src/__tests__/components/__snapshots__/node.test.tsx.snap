// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Node renders correctly 1`] = `
<div
  className="makeStyles-node-1"
  id="node-1"
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseUp={[Function]}
>
  <span>
    10
  </span>
</div>
`;
